diesel_cli/src/infer_schema_internals/sqlite.rs

```diff
@@ -134,6 +141,40 @@ pub fn get_table_data(
     // See: https://github.com/diesel-rs/diesel/issues/3579 as to why we use a direct
     // `sql_query` with `QueryableByName` instead of using `sql::<pragma_table_info::SqlType>`.
     let mut result = sql_query(query).load::<ColumnInformation>(conn)?;
+    // Add implicit rowid primary key column if the only primary key is rowid
+    // and ensure that the rowid column uses the right type.
+    let primary_key = get_primary_keys(conn, table)?;
+    if primary_key.len() == 1 {
+        let primary_key = primary_key.first().expect("guaranteed to have one element");
+        if !result.iter_mut().any(|x| {
+            if &x.column_name == primary_key {
+                if x.nullable && x.type_name.eq_ignore_ascii_case("Integer") {
+                    // A rowid column with type Integer was found, since rowid
+                    // is signed 64-bit integer, we should force that but for
+                    // now this gives type issues with foreign keys, so let's
+                    // leave it as is.
+                    x.type_name = String::from(ROWID_TYPE_NAME);
+                }
+                true
+            } else {
+                false
+            }
+        }) {
+            // Prepend implicit rowid column for the rowid implicit primary key.
+            result.insert(
+                0,
+                ColumnInformation {
+                    column_name: String::from(primary_key),
+                    type_name: String::from(ROWID_TYPE_NAME),
+                    type_schema: None,
+                    nullable: true,
```

Comment:
```suggestion
                    nullable: false,
```

I think rowid cannot ever by null, right? (At least that what's stated [here](https://sqlite.org/rowidtable.html))

Reply:
Good point - this was an intentional choice to make the auto generated column the same as the minimal explicitly created rowid which usually would also be defined nullable even though that it can never return NULL I think it can be set to NULL (and the database will then generate a new value) if that makes sense.

In my testing so far the generated schemas with this patch work just fine and no additional post processing is required for a functional database access to tables with implicit rowid.

Reply:
@weiznich I would prefer to keep the rowid column nullable for now, mostly to avoid too big of a change for existing Diesel users (e.g. https://github.com/diesel-rs/diesel/blob/1c24bf64ea2c4961d7d042c21e93e69fdb56073a/diesel_cli/tests/print_schema/print_schema_simple/sqlite/expected.snap#L17 would change).

Imo time will tell if this decision should be revisited.